### ISSUE TITLE
Command-line verification cleanup

### ISSUE THREAD

**Issue by dtowell** [logisim-evolution/logisim-evolution#1546]
The existing command-line verification features appears to be somehow haphazard. The documentation is at least incomplete and possibly misleading. Comments in the code suggest the existing command-line options are not organized well. Some options affect other options in ways only visible by reading the code. 

I am interested in rewriting (refactoring) the existing verification functionality (as I use it regularly). My priorities would be to improve (English) documentation, untangle the command-line options, and generally make the features useful for non-GUI verification tasks (like homework checking). In addition the code seems to suffer from feature creep and generally needs some clean up. A good solution would probably necessitate some command-line-options breaking changes. 

Is this worth pursuing? Are there non-negotiable constraints I might not be aware of?

**Comment by R3dst0ne** [comment]
A year ago, we partly reworked the command line options.
Have a look at the issues #963, #981 and the PR #983.

The discussion #598 is about improving the test vector component to add better support for sequential circuits testing.

The bad state of the documentation is a problem in general. So every help with it is more than welcome.

Before starting this endeavor, it might be a good to discuss this topic and collect ideas from people that use it / are interested in using it.

**Comment by maehne** [comment]
@dtowell: Help with improving the documentation and the verification functionality is very welcome! I agree with @R3dst0ne that it would be good if you would detail here your planned changes in more detail so that we can provide feedback before you start with the implementation.

**Comment by dtowell** [comment]
@maehne et al, I will be happy to do that. I don't mean to assume blanket coverage for any change. Once I investigate a bit further I will post a detailed plan (or at least goal).

**Comment by R3dst0ne** [comment]
Found an additional issue about the command line options.
#986

**Comment by maehne** [comment]
@dtowell: Thanks for your analysis and willingness to clean up these problems! I am looking forward to your PR(s)!

**Comment by dtowell** [comment]
@maehne @R3dst0ne please look over #1563 and provide feedback. I realize this is a large change. Does this look promising? Or, should I backup and try again?

**Comment by maehne** [comment]
@dtowell: PR #1563 goes for me in the right direction.

**Comment by dtowell** [comment]
#1581 should be good to review and hopefully integrate.

**Comment by hewzhew** [comment]
I rechecked this issue against the current `main` and the closed #1581 work. The broad refactor was never merged, and the parse-time exit-status inconsistency is still reproducible: invalid `--tty`, `--gates`, and `--geometry` values log errors but exit with status 0, while an unknown option or a missing TTY file exits nonzero.

I would be interested in a focused first slice that only distinguishes normal parser termination (`--help` / `--version`) from parser validation errors, routes those validation failures through one existing nonzero argument-error status, and adds focused regression coverage.

I would keep option-combination changes, GUI/TTY restructuring, runtime simulation/test statuses, and the wider `System.exit` cleanup out of this PR. Would this bounded scope be welcome?

**Comment by dtowell** [comment]
When the PR process stalled (or whatever) I "moved on". I no longer have time to spend on this project. I would be happy for anyone to pick up the pieces and make whatever changes they felt were desirable.

**Comment by hewzhew** [comment]
Thanks for clarifying, and for the work you put into #1563 and #1581. I’ll treat those PRs as useful references and keep any renewed work focused and independently validated.

**Comment by hewzhew** [comment]
After merged PR #2836, I found one remaining parse-time exit-status inconsistency: an invalid `--locale` still calls `System.exit(-1)` directly instead of using the existing argument-error status 10.

I have prepared a narrow follow-up that routes only this validation failure through the existing parser error path, while preserving the localized diagnostic, supported-locale list, valid-locale preloading, and localized help. It does not change runtime TTY/simulation statuses, option combinations, or other exit paths. Focused tests, the complete check, and a built-JAR exit-code matrix pass.

**Comment by hewzhew** [comment]
After #2836 and #2850, I would like to take a separate documentation-only third slice. The localized `pref-cmdline.html` pages still advertise removed options such as `-plain`, `-empty`, `-accents`, `-clearprops`, `-nosplash`, `-analyze`, `-testvector`, and `-geom`, omit current options, and list the Japanese locale as `jp` rather than `ja`.

I propose to update the English and Simplified Chinese pages to match the 17 options currently constructed by `Startup`. To avoid preserving inaccurate translations, the German, French, Portuguese, and Russian JavaHelp map entries for this page would temporarily fall back to the updated English page while retaining their localized TOC titles. CLI parsing, option behavior, and exit statuses would remain unchanged. Validation would cover the actual `--help` output, JavaHelp structure and links, and packaged HelpSets.

**Comment by maehne** [comment]
@hewzhew: Could you please have a look at closed PR #1546 by @dtowell and see what can be integrated into the current code base? Unfortunately, it got stuck at some point in the review process even though only minor things were left to be resolved. I think the work done there is still valuable and hope that it has not too many conflicts with the many recent changes.

**Comment by hewzhew** [comment]
Thanks for the pointer. I believe the intended closed PR is #1581, which superseded the Draft #1563. I compared its final head with current main. The whole branch is now 2491 commits behind and its 24-file patch overlaps heavily with today's `Main`, `Startup`, `TtyInterface`, `Download`, and locale resources, so rebasing or cherry-picking it wholesale would be unsafe.

The parse-time status part is already covered by merged #2836 and #2850. The most reusable remaining ideas are making `TtyInterface` return a status instead of calling `System.exit()` internally, using symbolic statuses, separating startup modes, and restoring runtime test coverage. However, #1581 also changes observable behavior: it uses argument status 1 instead of the current 10 and rejects several combinations that current main accepts, including GUI substitutions, multiple TTY files, GUI-only options with TTY, and mixed TTY/test modes.

I propose first taking a behavior-preserving slice that only centralizes TTY exit ownership and adds focused tests while retaining current numeric statuses and accepted combinations. A second slice could separate startup modes, still without policy changes. We could then discuss each option-combination restriction separately, and keep the command-line documentation update independent. Ready PR #2854 overlaps `Startup`/download handling, so I would account for its exact cable-selection contract before implementing the first slice. Does this sequence match what you had in mind?

**Comment by maehne** [comment]
@hewzhew: Your suggested approach makes sense to me.
